File: network/src/main/java/com/app/network/errorhandlers/OfflineException.java
package com.app.network.errorhandlers;

import java.io.IOException;


public final class OfflineException extends IOException {
  private final Throwable throwable;

  public OfflineException(final Throwable throwable) {
    super("");
    this.throwable = throwable;
  }

  public OfflineException(final String message, final Throwable throwable) {
    super(message);
    this.throwable = throwable;
  }


}

File: network/src/main/java/com/app/network/errorhandlers/OfflineException.kt
package com.app.network.errorhandlers

import com.app.jenkins_common.model.BaseModel
import com.app.jenkins_common.utils.EventBus
import com.squareup.otto.ThreadEnforcer


class OfflineException(private val throwable: Throwable) : BaseException(throwable) {

    private fun broadcastEvent() {
        EventBus.getInstance(ThreadEnforcer.ANY)
                .getBus()?.post(OfflineEvent(throwable.message, throwable.cause))
    }

    data class OfflineEvent(var message: String?, var throwable: Throwable?) : BaseModel(-1) {
        override fun className(): String {
            return this.className();
        }
    }
}
